source/server/converters.h

```diff
@@ -322,25 +321,21 @@ typename TypeToStorageType<TDriverType, TGrpcType>::StorageType allocate_output_
 
 const int kMaxGrpcErrorDescriptionSize = 2048;
 
-inline ::grpc::Status ApiErrorAndDescriptionToStatus(int32_t status, std::string& description, nlohmann::json& jsonError)
+template <typename TServerContext>
+inline ::grpc::Status ApiErrorAndDescriptionToStatus(TServerContext* context, int32_t status, std::string& description)
 {
+  context->AddTrailingMetadata("ni-error", std::to_string(status));
   converters::trim_trailing_nulls(description);
-  jsonError["code"] = status;
-  jsonError["message"] = description;
-  return ::grpc::Status(grpc::StatusCode::UNKNOWN, nlohmann::to_string(jsonError));
+  return ::grpc::Status(grpc::StatusCode::UNKNOWN, description);
 }
 
-inline ::grpc::Status ApiErrorAndDescriptionToStatus(int32_t status, std::string& description)
-{
-  nlohmann::json jsonError;
-  return ApiErrorAndDescriptionToStatus(status, description, jsonError);
-}
-
-inline ::grpc::Status ApiErrorToStatus(int32_t status)
+template <typename TServerContext>
```

Comment:
Note to reviewers: This is templatized so as to allow unit tests to Mock the context.